app/controllers/forms_controller.rb

```diff
@@ -25,20 +24,17 @@ def mark_pages_section_completed
                         else
                           t("banner.success.form.pages_saved")
                         end
-      redirect_to form_path(@form), success: success_message
+      redirect_to form_path(current_form), success: success_message
     else
       @mark_complete_form.mark_complete = "false"
-      render "pages/index", status: :unprocessable_entity
+      @forms = policy_scope(Form) || []
+      render "pages/index", locals: { current_form: }, status: :unprocessable_entity
     end
   end
 
 private
 
-  def current_form
-    @current_form ||= Form.find(params[:form_id])
-  end
```

Comment:
This was moved to application.rb so it could be used in any controller we would like. In this instance its being reused in PagesController and all the various Pages::XXXXXControllers